### ISSUE TITLE
Error when using Home Assistant Discovery

### ISSUE THREAD

**Issue by bachya** [bachya/ecowitt2mqtt#125]
From @alexdelprete:

Then I enabled HA discovery and got an error, and it stops sending data to MQTT. If I disable HA discovery, I can see updates in the MQTT topic again.

```
DEBUG:ecowitt2mqtt:Using arguments: Namespace(log_level='DEBUG', mqtt_broker='mqtt.axel.dom', mqtt_port=1883, mqtt_username='xxxxxx', mqtt_password='xxxxx', mqtt_topic='ecowitt2mqtt/GW2000A', hass_discovery=True, hass_discovery_prefix='homeassistant', endpoint='/data/report', port=8080, raw_data=None, input_unit_system='imperial', output_unit_system='metric')
DEBUG:asyncio:Using selector: EpollSelector
DEBUG:ecowitt2mqtt:Received data from Ecowitt device: {'PASSKEY': 'XXXXXXX', 'stationtype': 'GW2000A_V2.1.4', 'runtime': '1050700', 'dateutc': '2022-04-27 21:46:16', 'tempinf': '74.5', 'humidityin': '52', 'baromrelin': '30.056', 'baromabsin': '29.695', 'tempf': '57.6', 'humidity': '69', 'winddir': '353', 'windspeedmph': '4.92', 'windgustmph': '7.16', 'maxdailygust': '16.78', 'solarradiation': '0.00', 'uv': '0', 'rainratein': '0.000', 'eventrainin': '0.000', 'hourlyrainin': '0.000', 'dailyrainin': '0.000', 'weeklyrainin': '0.000', 'monthlyrainin': '1.449', 'yearlyrainin': '4.622', 'rrain_piezo': '0.000', 'erain_piezo': '0.000', 'hrain_piezo': '0.000', 'drain_piezo': '0.016', 'wrain_piezo': '0.079', 'mrain_piezo': '3.000', 'yrain_piezo': '6.173', 'ws90cap_volt': '5.1', 'ws90_ver': '115', 'temp1f': '72.1', 'humidity1': '61', 'temp2f': '73.0', 'humidity2': '57', 'temp3f': '71.4', 'humidity3': '65', 'temp4f': '74.7', 'humidity4': '55', 'temp5f': '70.7', 'humidity5': '65', 'temp6f': '74.3', 'humidity6': '55', 'temp7f': '68.9', 'humidity7': '48', 'temp8f': '68.2', 'humidity8': '52', 'soilmoisture1': '53', 'soilmoisture2': '58', 'soilmoisture3': '60', 'soilmoisture4': '47', 'soilmoisture5': '47', 'pm25_ch1': '0.0', 'pm25_avg_24h_ch1': '0.0', 'tf_co2': '62.4', 'humi_co2': '59', 'pm25_co2': '3.9', 'pm25_24h_co2': '4.8', 'pm10_co2': '4.1', 'pm10_24h_co2': '5.2', 'co2': '476', 'co2_24h': '482', 'lightning_num': '158', 'lightning': '1', 'lightning_time': '1651086083', 'wh80batt': '3.28', 'batt1': '0', 'batt2': '0', 'batt3': '0', 'batt4': '0', 'batt5': '0', 'batt6': '0', 'batt7': '0', 'batt8': '0', 'soilbatt1': '1.3', 'soilbatt2': '1.3', 'soilbatt3': '1.3', 'soilbatt4': '1.3', 'soilbatt5': '1.3', 'pm25batt1': '2', 'wh57batt': '3', 'co2_batt': '6', 'wh90batt': '3.22', 'freq': '868M', 'model': 'GW2000A'}
DEBUG:ecowitt2mqtt:Wind Chill Temperature is only defined for temperatures at or below 50 F and wind speeds above 3 mph. (temperature: 57.6, wind speed: 4.92)
DEBUG:ecowitt2mqtt:Publishing according to Home Assistant MQTT Discovery standard
DEBUG:ecowitt2mqtt:Sending CONNECT (u1, p1, wr0, wq0, wf0, c1, k60) client_id=b''
DEBUG:ecowitt2mqtt:Received CONNACK (0, 0)
INFO:ecowitt2mqtt:No entity description found for key: runtime
INFO:ecowitt2mqtt:No entity description found for key: runtime
DEBUG:ecowitt2mqtt:Sending DISCONNECT
ERROR:aiohttp.server:Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/src/ecowitt2mqtt/mqtt.py", line 114, in async_publish_payload
    await _async_publish_to_hass_discovery(client, data, discovery_manager)
  File "/usr/src/ecowitt2mqtt/mqtt.py", line 36, in _async_publish_to_hass_discovery
    _generate_payload("online"),
  File "/usr/src/ecowitt2mqtt/mqtt.py", line 77, in _generate_payload
    return value.encode("utf-8")
UnboundLocalError: local variable 'value' referenced before assignment
INFO:aiohttp.access:10.1.10.160 [27/Apr/2022:21:46:16 +0000] "POST /data/report HTTP/1.1" 500 244 "-" "-"
/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py:449: RuntimeWarning: coroutine 'Client.publish' was never awaited
  reset = await self.finish_response(request, resp, start_time)
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

_Originally posted by @alexdelprete in https://github.com/bachya/ecowitt2mqtt/issues/120#issuecomment-1111522066_

**Comment by alexdelprete** [comment]
@bachya still having issues with 2022.04.3 and HA discovery turned on:

```
DEBUG:ecowitt2mqtt:Received data from Ecowitt device: {'PASSKEY': 'C9E3FFE68DC4E12DEB7C96994872057B', 'stationtype': 'GW2000A_V2.1.4', 'runtime': '1129695', 'dateutc': '2022-04-28 19:42:49', 'tempinf': '75.2', 'humidityin': '48', 'baromrelin': '30.073', 'baromabsin': '29.713', 'tempf': '61.9', 'humidity': '39', 'winddir': '6', 'windspeedmph': '10.51', 'windgustmph': '19.68', 'maxdailygust': '29.53', 'solarradiation': '0.00', 'uv': '0', 'rainratein': '0.000', 'eventrainin': '0.000', 'hourlyrainin': '0.000', 'dailyrainin': '0.000', 'weeklyrainin': '0.000', 'monthlyrainin': '1.449', 'yearlyrainin': '4.622', 'rrain_piezo': '0.000', 'erain_piezo': '0.000', 'hrain_piezo': '0.000', 'drain_piezo': '0.000', 'wrain_piezo': '0.079', 'mrain_piezo': '3.000', 'yrain_piezo': '6.173', 'ws90cap_volt': '5.2', 'ws90_ver': '115', 'temp1f': '73.2', 'humidity1': '56', 'temp2f': '73.9', 'humidity2': '54', 'temp3f': '72.9', 'humidity3': '57', 'temp4f': '75.6', 'humidity4': '51', 'temp5f': '72.0', 'humidity5': '60', 'temp6f': '75.0', 'humidity6': '52', 'temp7f': '70.7', 'humidity7': '32', 'temp8f': '71.8', 'humidity8': '37', 'soilmoisture1': '55', 'soilmoisture2': '59', 'soilmoisture3': '62', 'soilmoisture4': '51', 'soilmoisture5': '52', 'pm25_ch1': '0.0', 'pm25_avg_24h_ch1': '0.0', 'tf_co2': '63.1', 'humi_co2': '39', 'pm25_co2': '5.2', 'pm25_24h_co2': '6.2', 'pm10_co2': '6.0', 'pm10_24h_co2': '6.9', 'co2': '526', 'co2_24h': '483', 'lightning_num': '0', 'lightning': '1', 'lightning_time': '1651086083', 'wh80batt': '3.28', 'batt1': '0', 'batt2': '0', 'batt3': '0', 'batt4': '0', 'batt5': '0', 'batt6': '0', 'batt7': '0', 'batt8': '0', 'soilbatt1': '1.3', 'soilbatt2': '1.3', 'soilbatt3': '1.3', 'soilbatt4': '1.3', 'soilbatt5': '1.3', 'pm25batt1': '2', 'wh57batt': '2', 'co2_batt': '6', 'wh90batt': '3.22', 'freq': '868M', 'model': 'GW2000A'}
DEBUG:ecowitt2mqtt:Wind Chill Temperature is only defined for temperatures at or below 50 F and wind speeds above 3 mph. (temperature: 61.9, wind speed: 10.51)
DEBUG:ecowitt2mqtt:Publishing according to Home Assistant MQTT Discovery standard
DEBUG:ecowitt2mqtt:Sending CONNECT (u1, p1, wr0, wq0, wf0, c1, k60) client_id=b''
DEBUG:ecowitt2mqtt:Received CONNACK (0, 0)
INFO:ecowitt2mqtt:No entity description found for key: runtime
INFO:ecowitt2mqtt:No entity description found for key: runtime
DEBUG:ecowitt2mqtt:Sending DISCONNECT
ERROR:aiohttp.server:Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.9/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/src/ecowitt2mqtt/mqtt.py", line 114, in async_publish_payload
    await _async_publish_to_hass_discovery(client, data, discovery_manager)
  File "/usr/src/ecowitt2mqtt/mqtt.py", line 36, in _async_publish_to_hass_discovery
    _generate_payload("online"),
  File "/usr/src/ecowitt2mqtt/mqtt.py", line 77, in _generate_payload
    return value.encode("utf-8")
UnboundLocalError: local variable 'value' referenced before assignment
INFO:aiohttp.access:10.1.10.160 [28/Apr/2022:19:42:49 +0000] "POST /data/report HTTP/1.1" 500 244 "-" "-"
/usr/local/lib/python3.9/site-packages/aiohttp/web_protocol.py:449: RuntimeWarning: coroutine 'Client.publish' was never awaited
  reset = await self.finish_response(request, resp, start_time)
RuntimeWarning: Enable tracemalloc to get the object allocation traceback

```

**Comment by alexdelprete** [comment]
@bachya confirmed working:

```
INFO:ecowitt2mqtt:Published to HASS discovery: {'runtime': 1131342, 'tempin': 24.0, 'humidityin': 49, 'baromrel': 1018.592, 'baromabs': 1006.401, 'temp': 16.1, 'humidity': 41, 'winddir': 10, 'windspeed': 10.4, 'windgust': 18.4, 'maxdailygust': 29.53, 'solarradiation': 0.0, 'uv': 0, 'rainrate': 0.0, 'eventrain': 0.0, 'hourlyrain': 0.0, 'dailyrain': 0.0, 'weeklyrain': 0.0, 'monthlyrain': 36.8, 'yearlyrain': 117.4, 'rrain_piezo': 0.0, 'erain_piezo': 0.0, 'hrain_piezo': 0.0, 'drain_piezo': 0.0, 'wrain_piezo': 2.0, 'mrain_piezo': 76.2, 'yrain_piezo': 156.8, 'ws90cap_volt': 5.2, 'ws90_ver': 115, 'temp1': 22.9, 'humidity1': 56, 'temp2': 23.4, 'humidity2': 53, 'temp3': 22.8, 'humidity3': 59, 'temp4': 24.2, 'humidity4': 51, 'temp5': 22.2, 'humidity5': 60, 'temp6': 23.9, 'humidity6': 53, 'temp7': 21.3, 'humidity7': 34, 'temp8': 21.8, 'humidity8': 38, 'soilmoisture1': 55, 'soilmoisture2': 59, 'soilmoisture3': 62, 'soilmoisture4': 51, 'soilmoisture5': 52, 'pm25_ch1': 0.0, 'pm25_avg_24h_ch1': 0.0, 'tf_co2': 62.8, 'humi_co2': 40, 'pm25_co2': 6.2, 'pm25_24h_co2': 6.2, 'pm10_co2': 6.6, 'pm10_24h_co2': 6.9, 'co2': 475, 'co2_24h': 483, 'lightning_num': 0.0, 'lightning': 0.6, 'lightning_time': datetime.datetime(2022, 4, 27, 19, 1, 23, tzinfo=datetime.timezone.utc), 'wh80batt': 3.28, 'batt1': 'OFF', 'batt2': 'OFF', 'batt3': 'OFF', 'batt4': 'OFF', 'batt5': 'OFF', 'batt6': 'OFF', 'batt7': 'OFF', 'batt8': 'OFF', 'soilbatt1': 1.3, 'soilbatt2': 1.3, 'soilbatt3': 1.3, 'soilbatt4': 1.3, 'soilbatt5': 1.3, 'pm25batt1': 'ON', 'wh57batt': 'ON', 'co2_batt': 'ON', 'wh90batt': 3.22, 'dewpoint': 2.9, 'feelslike': 16.1, 'heatindex': 14.8, 'solarradiation_lux': 0.0, 'solarradiation_perceived': 0.0, 'windchill': None}
```

Aaron, I noticed the digests for the images of tags latest and 2022.4.3 are different, is it correct?

![image](https://user-images.githubusercontent.com/7027842/165837521-8ade13e5-d8ce-41a5-9ed4-b94b2d2b3617.png)

**Comment by bachya** [comment]
@alexdelprete Nothing to be concerned about.
